### PR TITLE
Correctly handle pull requests from master branch

### PR DESCRIPTION
**Pull request by marcinkunert**
This should fix (or ad least improve) the situation in which a pull request is created from the fork master branch as described here #1998

### CODE REVIEW

**Comment by marcinkunert** [comment]
I've also changed so that the script will be run from upstream master, not the fork. This should fix situations like this fail: #2006

File: .github/workflows/translation_check.yml [hunk 1]
```diff
@@ -30,20 +30,48 @@ jobs:
...
 
+      - name: Checkout upstream master (in order to use the newest version of the script)
+        uses: actions/checkout@v2
+        with:
+          path: upstream_master
+          ref: master
+
       - name: Run checks
-        run: python master/.github/workflows/translation_check.py
+        run: python upstram_master/.github/workflows/translation_check.py
```
Comment (tupaschoal) [inline .github/workflows/translation_check.yml:74]: There is a typo here "upstram"

File: .github/workflows/translation_check.yml [hunk 2]
```diff
@@ -30,20 +30,48 @@ jobs:
...
 
+      - name: Print debug stuff
+        run: |
+          echo ${{steps.merge-base-commit.outputs.merge_base}}
+          echo ${{github.event.pull_request.head.sha}}
+          echo ${{steps.merge-base-commit.outputs.merge_base == github.event.pull_request.head.sha}}
+
+      - name: Remove master
+        if: ${{ steps.merge-base-commit.outputs.merge_base == github.event.pull_request.head.sha }}
+        run: rm -rf master
```
Comment (tupaschoal) [inline .github/workflows/translation_check.yml:53]: Not sure we should force it

File: .github/workflows/translation_check.yml (hunk 1 above)
Comment (marcinkunert) [reply .github/workflows/translation_check.yml:74]: That's true :( Fixed in next PR

File: .github/workflows/translation_check.yml (hunk 2 above)
Comment (marcinkunert) [reply .github/workflows/translation_check.yml:53]: This doesn't matter as it only will be deleted in some cases and the directory will be created again in the next step

**Pull request merged**